src/content/en/showcase/2017/with.in.md

```diff
@@ -0,0 +1,331 @@
+   </td>
+   <td>X
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+   <td>
+   </td>
+  </tr>
+  <tr>
+   <td><a href="https://vr.with.in/archive/text-sdf-bitmap/">Signed Distance Field Bitmap Text</a>
+   </td>
+   <td>X
+   </td>
+   <td>X
+   </td>
+   <td>X
+   </td>
+   <td>
+   </td>
+  </tr>
+</table>
+
+
+## Our Decision: SDF Bitmap Font
+
+2D canvas with `ctx.fillText()` can do text wrapping, letter spacing and line height, but overflow gets cut off, and text will be blurry if you zoom in really far. You could increase the size of the canvas texture, but might hit an upper limit in texture size or performance could suffer if the texture is too big.
+
+Extruded 3D text is essentially the same as triangulated vector text, but with depth and possibly a bevel, so it has at least twice as much geometry. Either of these could work in small doses for titles or logos, but wouldn't perform as well for large quantities of text, and neither has typographic features.
+
+Bitmap fonts use one quad (two triangles) per character, so it uses less geometry and performs better than [triangulated vectors](https://vr.with.in/archive/text-triangulated-vector/). They're still raster based since they use a texture map sprite, but with an SDF shader, they're basically resolution independent, so they look nicer than a 2D canvas texture. Matt DesLauriers' three-bmfont-text also includes reliable typographic features for text wrapping, letter spacing, line height and alignment. Overflow doesn't get cut off. Font size is controlled through scale. We chose this route because it gave us the best options for design while still staying performant. Unfortunately, it wasn't as easy to implement so we'll go through the steps in hopes to alleviate fellow developers working in WebVR.
+
+[mattdesl.svbtle.com/material-design-on-the-gpu](https://mattdesl.svbtle.com/material-design-on-the-gpu)
+
+[mattdesl.github.io/physical-text/index.html](http://mattdesl.github.io/physical-text/index.html)
+
+[github.com/mattdesl/physical-text](https://github.com/mattdesl/physical-text)
+
+
+<figure>
+  <img src="/web/showcase/2017/images/with.in/diagram.png">
+  <figcaption>Font to SDF Bitmap Workflow</figcaption>
+</figure>
+
+
+## 1. Generate bitmap font (.png + .fnt)
+
+[libgdx.badlogicgames.com/tools.html](https://libgdx.badlogicgames.com/tools.html)
+
+[github.com/libgdx/libgdx/wiki/Hiero](https://github.com/libgdx/libgdx/wiki/Hiero)
+
+[github.com/libgdx/libgdx/wiki/Distance-field-fonts](https://github.com/libgdx/libgdx/wiki/Distance-field-fonts)
+
+Hiero is a bitmap font packing tool that runs with Java. The Hiero documentation doesn't really explain how to run it without going through a complicated build process. First, install Java if you haven't already. Then if double clicking on the runnable-hiero.jar doesn't open Hiero, try running it with this command in the console:
+
+
+```
+java -jar runnable-hiero.jar
```

Comment:
**Strong preference** for indenting with 4 spaces instead of wrapping with three backticks.